### PR TITLE
Remove procfs dependency

### PR DESCRIPTION
**Pull request by tamird**
- Remove verifier log special case
- Remove procfs dependency
- Rewrite kernel version logic

This restores and enhances the logic originally added in #579.

### CODE REVIEW

File: aya/src/util.rs [hunk 1]
```diff
@@ -13,9 +13,130 @@ use crate::{
...
+    pub fn new(major: u8, minor: u8, patch: u16) -> Self {
+        Self {
+            major,
+            minor,
+            patch,
+        }
+    }
+
+    /// Returns the kernel version of the currently running kernel.
+    pub fn current() -> Result<Self, String> {
```
Comment (alessandrod) [inline aya/src/util.rs:38]: I think we should return a `Box<dyn Error>` instead of String here

Comment (tamird) [reply aya/src/util.rs:38]: Can you help me understand how that's better? Currently aya does not depend on anyhow, so other than using strings I can't propagate context.

Comment (alessandrod) [reply aya/src/util.rs:38]: This is now a public API, and returning String as Err is frowned upon (i think there's even a clippy lint for it)

> Currently aya does not depend on anyhow, so other than using strings I can't propagate context.

libraries should depend on thiserror, apps on anyhow. You can define a thiserror derived error, give it one variant for each string you're returning now, and type erase it when you return it. The result is the same as using a string (Error implements Display), but it's not a String, gives you a backtrace on nightly if you want one etc

Comment (tamird) [reply aya/src/util.rs]: Done.

**Review by alessandrod — APPROVED** [review]

**Pull request merged**